synapse/storage/databases/main/event_push_actions.py

```diff
@@ -182,6 +182,7 @@ class UserPushAction(EmailPushAction):
     profile_tag: str
 
 
+# TODO This is used as a cached value and is mutable.
```

Comment:
Hmm, I take it something is mutating it?

Reply:
We instantiate a `NotifCounts` and then increment it based on 3 different queries. I'm sure we can refactor a bit, but I figured we could clean that up after the fact.

See `_get_unread_counts_by_pos_txn`:

https://github.com/matrix-org/synapse/blob/eee2b6642ddb28a6e0f850b958baa39eb74dc891/synapse/storage/databases/main/event_push_actions.py#L639-L641